File: freecodecamp/2026-01-10.js
function ticTacToe(board) {
    const lines = [
        // rows
        [[0, 0], [0, 1], [0, 2]],
        [[1, 0], [1, 1], [1, 2]],
        [[2, 0], [2, 1], [2, 2]],

        // columns
        [[0, 0], [1, 0], [2, 0]],
        [[0, 1], [1, 1], [2, 1]],
        [[0, 2], [1, 2], [2, 2]],

        // diagonals
        [[0, 0], [1, 1], [2, 2]],
        [[0, 2], [1, 1], [2, 0]]
    ];

    for (const line of lines) {
        const [a, b, c] = line;
        const value = board[a[0]][a[1]];

        if (
            value === board[b[0]][b[1]] &&
            value === board[c[0]][c[1]]
        ) {
            return value === "X" ? "X wins" : "O wins";
        }
    }

    return "Draw";
}

File: freecodecamp/2026-01-10.py
def tic_tac_toe(board):
    lines = [
        # rows
        [(0, 0), (0, 1), (0, 2)],
        [(1, 0), (1, 1), (1, 2)],
        [(2, 0), (2, 1), (2, 2)],

        # columns
        [(0, 0), (1, 0), (2, 0)],
        [(0, 1), (1, 1), (2, 1)],
        [(0, 2), (1, 2), (2, 2)],

        # diagonals
        [(0, 0), (1, 1), (2, 2)],
        [(0, 2), (1, 1), (2, 0)]
    ]

    for line in lines:
        (r1, c1), (r2, c2), (r3, c3) = line
        value = board[r1][c1]

        if value == board[r2][c2] == board[r3][c3]:
            return "X wins" if value == "X" else "O wins"

    return "Draw"

